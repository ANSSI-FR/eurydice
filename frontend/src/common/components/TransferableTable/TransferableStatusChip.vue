<template>
  <v-tooltip color="black" right>
    <template #activator="{ on }">
      <v-chip text-color="white" :color="color" v-on="on">
        <v-progress-circular
          v-if="state === 'ONGOING'"
          :value="progress"
          :width="7"
          size="17"
          class="mr-2"
        ></v-progress-circular
        ><v-icon v-else class="mr-2" size="17">{{ icon }}</v-icon
        >{{ stateLabel }}
      </v-chip>
    </template>
    <span>{{ tooltipText }}</span>
  </v-tooltip>
</template>
<script>
export default {
  name: "TransferableStatusChip",
  props: {
    state: {
      type: String,
      required: true,
    },
    stateLabel: {
      type: String,
      required: true,
    },
    progress: {
      type: Number,
      required: false,
      default: 100,
    },
    icon: {
      type: String,
      required: true,
    },
    color: {
      type: String,
      required: true,
    },
    tooltipText: {
      type: String,
      required: true,
    },
  },
  mounted() {
    if (this.state === "SUCCESS") {
      this.$store.commit("setOngoingTransfer", false);
    }
  },
};
</script>
<style lang="scss">
.v-progress-circular__underlay {
  display: none;
}
</style>
