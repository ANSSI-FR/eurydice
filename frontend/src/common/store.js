import Vue from "vue";
import Vuex from "vuex";

Vue.use(Vuex);

export default new Vuex.Store({
  state: () => ({
    username: null,
    maintenance: false,
    serverDown: false,
    serverMetadata: null,
    ongoingTransfer: false,
  }),
  getters: {
    maintenance(state) {
      return state.maintenance;
    },
    serverDown(state) {
      return state.serverDown;
    },
    serverMetadata(state) {
      return state.serverMetadata;
    },
    ongoingTransfer(state) {
      return state.ongoingTransfer;
    },
  },
  mutations: {
    setUsername(state, newUsername) {
      state.username = newUsername;
    },
    setMaintenance(state, newMaintenance) {
      state.maintenance = newMaintenance;
    },
    setServerDown(state, serverDown) {
      state.serverDown = serverDown;
    },
    setServerMetadata(state, serverMetadata) {
      state.serverMetadata = serverMetadata;
    },
    setOngoingTransfer(state, ongoingTransfer) {
      state.ongoingTransfer = ongoingTransfer;
    },
  },
  actions: {},
  // NOTE: consider adding modules to this common store in main.js
  //       when adding origin/destination specific state/mutations
  modules: {},
});
