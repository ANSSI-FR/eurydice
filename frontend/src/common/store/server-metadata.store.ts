import * as serverMetadataService from '@common/services/server-metadata.service';
import { defineStore } from 'pinia';
import { computed, ref } from 'vue';

export const useServerMetadataStore = defineStore('serverMetadataStore', () => {
  const serverMetadata = ref<ServerMetadata>();

  const getServerMetadata = computed(() => serverMetadata.value);
  const setServerMetadata = (metadata: ServerMetadata | undefined) => {
    serverMetadata.value = metadata;
  };
  const fetchServerMetadata = async () => {
    await serverMetadataService.getServerMetadata().then((fetchedServerMetadata) => {
      setServerMetadata(fetchedServerMetadata);
    });
  };
  const $reset = () => {
    serverMetadata.value = undefined;
  };
  return {
    getServerMetadata,
    setServerMetadata,
    fetchServerMetadata,
    $reset,
  };
});
