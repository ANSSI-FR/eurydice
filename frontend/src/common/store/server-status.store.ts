import { defineStore } from 'pinia';
import { computed, ref } from 'vue';

export const useServerStatusStore = defineStore('serverStatusStore', () => {
  const isServerDown = ref<boolean>(false);
  const isServerInMaintenance = ref<boolean>(false);

  const getIsServerDown = computed(() => isServerDown.value);
  const setIsServerDown = (value: boolean) => {
    isServerDown.value = value;
  };

  const getIsServerInMaintenance = computed(() => isServerInMaintenance.value);
  const setIsServerInMaintenance = (value: boolean) => {
    isServerInMaintenance.value = value;
  };

  const $reset = () => {
    isServerDown.value = false;
    isServerInMaintenance.value = false;
  };
  return {
    getIsServerDown,
    setIsServerDown,
    getIsServerInMaintenance,
    setIsServerInMaintenance,
    $reset,
  };
});
