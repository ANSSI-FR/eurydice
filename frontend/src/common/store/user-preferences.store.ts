import type { UserPreferences } from '@common/models/UserPreferences';
import { defineStore } from 'pinia';
import { readonly, ref } from 'vue';

export const useUserPreferencesStore = defineStore('userPreferencesStore', () => {
  const LOCAL_STORAGE_KEY: string = 'userPreferences';

  const updateDarkModeInHTML = (darkMode: boolean | undefined) => {
    if (darkMode) {
      document.querySelector('html')!.classList.add('dark');
    } else {
      document.querySelector('html')!.classList.remove('dark');
    }
  };

  const initFromLocalStorage = () => {
    let preferences;
    const localStoragePreferences = localStorage.getItem(LOCAL_STORAGE_KEY);
    if (localStoragePreferences) {
      preferences = JSON.parse(localStoragePreferences);
    } else {
      const systemDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches;
      preferences = { darkMode: systemDarkMode };
    }
    updateDarkModeInHTML(preferences.darkMode);
    return preferences;
  };

  const saveUserPreferencesToLocalStorage = () => {
    localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(userPreferences.value));
  };

  const userPreferences = ref<UserPreferences | undefined>(initFromLocalStorage());

  const getUserPreferences = readonly(userPreferences);

  const setDarkModePreference = (darkMode: boolean) => {
    if (!userPreferences.value) {
      userPreferences.value = {};
    }
    if (userPreferences.value.darkMode !== darkMode) {
      userPreferences.value.darkMode = darkMode;
      saveUserPreferencesToLocalStorage();
      updateDarkModeInHTML(darkMode);
    }
  };

  return {
    getUserPreferences,
    setDarkModePreference,
  };
});
