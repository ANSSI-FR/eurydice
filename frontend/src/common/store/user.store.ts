import { defineStore } from 'pinia';
import { computed, ref } from 'vue';

export const useUserStore = defineStore('userStore', () => {
  const currentUser = ref<User>();

  const getCurrentUser = computed(() => currentUser.value);
  const setCurrentUser = (user: User | undefined) => {
    currentUser.value = user;
  };
  const $reset = () => {
    currentUser.value = undefined;
  };
  return {
    getCurrentUser,
    setCurrentUser,
    $reset,
  };
});
