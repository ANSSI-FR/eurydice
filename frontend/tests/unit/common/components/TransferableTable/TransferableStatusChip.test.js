import TransferableStatusChip from "@common/components/TransferableTable/TransferableStatusChip";
import { createLocalVue, mount } from "@vue/test-utils";
import Vuetify from "vuetify";
import store from "@/common/store";

describe("TransferableStatusChip", () => {
  const localVue = createLocalVue();
  const vuetify = new Vuetify();

  const tooltipText = "The world is a sphere. There is no East or West.";
  const wrapper = mount(TransferableStatusChip, {
    localVue,
    vuetify,
    store,
    propsData: {
      state: "SUCCESS",
      stateLabel: "SuccÃ¨s",
      icon: "mdi-check",
      color: "success",
      tooltipText,
    },
  });
  it("renders correctly", async () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  it("displays given tooltip text", async () => {
    wrapper.findComponent(TransferableStatusChip).trigger("mouseenter");
    await wrapper.vm.$nextTick();

    requestAnimationFrame(() => {
      expect(wrapper.find("span").text()).toBe(tooltipText);
    });
  });
});
