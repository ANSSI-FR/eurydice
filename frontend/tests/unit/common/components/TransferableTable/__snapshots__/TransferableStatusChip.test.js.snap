// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransferableStatusChip renders correctly 1`] = `
<span
  class="v-tooltip v-tooltip--right"
>
  <!---->
  <span
    class="v-chip theme--light v-size--default success white--text"
  >
    <span
      class="v-chip__content"
    >
      <i
        aria-hidden="true"
        class="v-icon notranslate mr-2 mdi mdi-check theme--light"
        style="font-size: 17px;"
      />
      Succ√®s
    
    </span>
  </span>
</span>
`;
